### DISCUSSION TITLE
Set Metadata for every new excalidraw drawing

### DISCUSSION THREAD

**Discussion by tomspeed22** [zsviczian/obsidian-excalidraw-plugin#2094] [Q&A]
Hi,
Is there a way to automatically add a metada (custom made or aliases) when creating a new drawing in exclidraw for obsidian ? See example in attachment. Thanks
![Snag_19be1d65](https://github.com/user-attachments/assets/424e89b1-6d97-4cfe-9721-3e47fcbb2469)

**Comment by tomspeed22** [comment]
I have tried to add this code to the startup not from Excalidraw automation but it does not add the aliases. Thanks for help

``

if (typeof ea === 'undefined') {
    var ea = ExcalidrawAutomate;
}

// Hook to run when a new Excalidraw file is created
ea.onFileCreateHook = async (data) => {
    ea.reset();
    ea.addMetadata({
        "aliases": "yourAlias",
    });
    await ea.save();
};

// Function to create a new Excalidraw file
async function createNewExcalidrawFile() {
    const newFile = await ea.createDrawing(); // Adjust this line based on the correct method to create a new drawing
    console.log("New Excalidraw file created:", newFile);
}

// Call the function to create a new file
createNewExcalidrawFile();
``

**Comment by CodeFromRahul** [comment]
now their is an aditional functionality for the above problem they implemented that
